enumerables.rb

```diff
@@ -0,0 +1,139 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
+
+# Enumerable Module
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = *self
+    x = 0
+    while x < arr.size
+      yield(arr[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |element|
+      yield(element, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << i if yield(i)
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
+    flag = true
+    my_each do |i|
```

Comment:
- [ ] `my_all?` when no block or argument is given returns true when none of the collection members are false or nil.
```
Failure/Error: expect(true_array.my_all?).to be true_array.all?
     
       expected true
            got false

```
- Use this as your true_array `[1, true, 'hi', []]`. Then compare your enumerable method `my_all?` with the ruby enumerable method of `all?`. Make sure they both output the same outcomes.